app/models/event.rb

```diff
@@ -20,6 +20,62 @@ class Event < ApplicationRecord
     in_week(week).group_by { |e| e.start_time.wday }
   }
 
+  class << self
+    def not_free(week)
+      css_color = 'repeating-linear-gradient(-45deg,#D3E9FF,#D3E9FF 10px,#B8DCFF 10px,#B8DCFF 20px)'
+      occupied_slot = ConferenceRoom.new(capacity: 99, title: 'NoWayLand', color: css_color, email: 'fake_email@com')
+      Hash[(1..7).map { |n| [n, []] }].merge(occupied_slots_per_wday(order('start_time').in_week_group_by_weekday(week))).tap do |vals|
+        vals.each_value do |ranges|
+          ranges.map! do |range|
+            Event.new(start_time: range.begin, end_time: range.end, name: 'Impossibru', description: 'Occupied', user: 'Very occupied user', conference_room: occupied_slot)
+          end
+        end
+      end
+    end
+
+    def occupied_slots_per_wday(reservations)
+      occupied = {}
+      empty_rooms = Hash[ConferenceRoom.all.map { |c| [c, []] }]
+      reservations.each { |wday, events| reservations[wday] = empty_rooms.merge(events.group_by(&:conference_room)) }
+      reservations.each do |wday, rooms|
+        not_free = []
+        flag = true
+        rooms.each do |room, events|
+          if flag
+            events.each { |n| not_free << (n.start_time..n.end_time)}
+            flag = false
+          else
+            tmp = []
+            events.each do |event|
+              not_free.each { |i| (nf = intersection(i, event)) && tmp << nf }
+            end
+            not_free = tmp
+            break if tmp == []
+          end
+        end
+        occupied[wday] = not_free
+      end
+      occupied
+    end
+
+    def intersection(left, right)
+      return nil unless left && right
+      return nil if left.max < right.begin || right.max < left.begin
+      [left.begin, right.begin].max..[left.max, right.max].min
+    end
+  end
+
+
+  def begin
+    start_time
```

Comment:
Why you aliasing these methods? Can we use `alias_method` instead `def`? Also `begin` as method name is a bit controversial :D